okapi/services/caches/geocaches.php

```diff
@@ -703,17 +703,32 @@ public static function call(OkapiRequest $request)
             foreach ($results as &$result_ref)
                 $result_ref['latest_logs'] = array();
 
-            # Get all log IDs with dates. Sort in groups. Filter out latest ones. This is the fastest
-            # technique I could think of...
+            # Get all log IDs in proper order, then filter out the latest
+            # ones. This should be the fastest technique ...
+
+            # OCDE allows to submit logs without time. To minimize problems
+            # when ordering logs with and without time on the same day, there
+            # is a separate order_date field which is caluclated from 'date'
+            # and 'date_created'.
+            #
+            # OCPL log entries are all submitted with time, so the 'date'
+            # field is sufficient for ordering.
+
+            if (Settings::get('OC_BRANCH') == 'oc.de') {
+                $logs_order_field_SQL = 'order_date';
+            } else {
+                $logs_order_field_SQL = 'date';
+            }
 
             $rs = Db::query("
-                select cache_id, uuid, date
+                select cache_id, uuid
                 from cache_logs
                 where
                     cache_id in ('".implode("','", array_map('\okapi\Db::escape_string', array_keys($cacheid2wptcode)))."')
                     and ".((Settings::get('OC_BRANCH') == 'oc.pl') ? "deleted = 0" : "true")."
-                order by cache_id, date desc, date_created desc
+                order by cache_id, ".$logs_order_field_SQL." desc, date_created desc, id desc
```

Comment:
Copied from 3dbe476:

> I verified that this ordering is index-optimized on OCDE database, but _no_ index is used at OCPL. This was also the case for the original order statement. there is a cache_id/deleted/date/date_created index at OCPL, but it will only be used if `FORCE INDEX (by_cache)` is added.

Oh, so you're **sure** the index is not used in OCPL?